internal/docstore/driver/driver.go

```diff
@@ -104,10 +103,21 @@ type Filter struct {
 	Value     interface{} // the value to compare using the operation
 }
 
-// A DocumentIterator is an iterator that can iterator through the results (for Get
-// action), and returns the next page that can be passed into a Query as a
-// starting point.
+// A DocumentIterator iterates through the results (for Get action), and returns
+// the next page that can be passed into a Query as a starting point.
 type DocumentIterator interface {
-	Next(doc Document) error
+
+	// Next get the next item in the query result and calls decode to decode into
+	// Document.
+	//
+	// When it is called when the iterator is at the end of a page, it should go
```

Comment:
Again, not sure why you're talking about pages here. The interface doesn't use pages.

Reply:
Clarified, I was thinking that the driver implementation may do a query in a page size smaller than limit.